base/src/main/java/org/mozilla/jss/provider/javax/crypto/JSSTrustManager.java

```diff
@@ -175,25 +246,15 @@ public void checkKeyUsage(X509Certificate[] certChain, String keyUsage) throws E
             return;
         }
 
-        String msg = "Missing EKU: " + keyUsage +
-            ". Certificate with subject DN `" + cert.getSubjectX500Principal() + "` had ";
-
         if (extendedKeyUsages == null) {
-            msg += "no EKU extension";
+            logger.debug("JSSTrustManager: Missing extended key usage extension");
 
         } else {
-            msg += "EKUs { ";
-            boolean first = true;
-            for (String eku : extendedKeyUsages) {
-                if (!first) msg += " , ";
-                msg += eku;
-                first = false;
-            }
-            msg += " }";
+            logger.debug("JSSTrustManager: Missing " + keyUsage + " key usage");
         }
 
-        msg += ".  class = " + cert.getClass();
-        throw new CertificateException(msg);
+        // TODO: fix depth param
+        status.addReason(ValidityStatus.INADEQUATE_KEY_USAGE, cert, 0);
```

Comment:
This check is done only for the leaf certificate so the depth is correct. The TODO can be removed.

Reply:
Please see my comment below.